hexrdgui/image_canvas.py

```diff
@@ -1067,15 +1102,66 @@ def finish_show_polar(self, iviewer):
                 formatter = PolarXAxisFormatter(default_formatter, f)
                 axis.xaxis.set_major_formatter(formatter)
 
+                axis.yaxis.set_major_locator(AutoLocator())
+                axis.yaxis.set_minor_locator(AutoMinorLocator())
+
+                axis.xaxis.set_major_locator(AutoLocator())
+                axis.xaxis.set_minor_locator(AutoMinorLocator())
+
+                # change property of ticks
+                kwargs = {
+                    'left': True,
+                    'right': True,
+                    'bottom': True,
+                    'top': True,
+                    'which': 'major',
+                    'length': 10,
+                    'labelfontfamily': 'serif',
+                    'labelsize': FONTSIZE_TICKS,
+                }
+                axis.tick_params(**kwargs)
+
+                kwargs['which'] = 'minor'
+                kwargs['length'] = 2
+                axis.tick_params(**kwargs)
+
+                # add grid lines parallel to x-axis in azimuthal average
+                kwargs = {
+                    'visible': True,
+                    'which': 'major',
+                    'axis': 'y',
+                    'linewidth': 0.25,
+                    'linestyle': '-',
+                    'color': 'k',
+                    'alpha': 0.75,
+                }
+                axis.grid(**kwargs)
+
+                kwargs = {
+                    'visible': True,
+                    'which': 'minor',
+                    'axis': 'y',
+                    'linewidth': 0.075,
+                    'linestyle': '--',
+                    'color': 'k',
+                    'alpha': 0.9,
+                }
+                axis.grid(**kwargs)
+
+                # add grid lines parallel to y-axis
+                kwargs['which'] = 'both'
+                kwargs['axis'] = 'x'
+                axis.grid(**kwargs)
+
                 # Set our custom tick locators as well
-                self.axis.xaxis.set_major_locator(PolarXAxisTickLocator(self))
-                axis.xaxis.set_major_locator(PolarXAxisTickLocator(self))
+                # self.axis.xaxis.set_major_locator(PolarXAxisTickLocator(self))
+                # axis.xaxis.set_major_locator(PolarXAxisTickLocator(self))
```

Comment:
So, I think we were using these before so that when you set the x-axis to use Q instead, tick markers show up at `1, 2, 3, 4, ...`. Can you verify that still happens with the above changes? If not, we might need to fix that before merging.

Reply:
Yes, looks like switching to `Q` messes up the `xticklabels`. We'd have to redo this step when the axis is updated to use `Q`.

<img width="452" alt="Screenshot 2024-11-11 at 2 57 17 PM" src="https://github.com/user-attachments/assets/f1200f08-21c4-4422-b48b-1acbf769ed73">
<img width="446" alt="Screenshot 2024-11-11 at 2 57 32 PM" src="https://github.com/user-attachments/assets/a4f81bb6-e84c-4964-a1a8-4d40320c9749">

Reply:
So actually, those `PolarXAxisTickLocator` classes inherit `AutoLocator`. I think that in the code above, where you have `yaxis.set_major_locator(AutoLocator())`, we might be able to just change that `AutoLocator` to `PolarXAxisTickLocator`.

However, we probably also need to change the minor locator, too.